pkgs/development/python-modules/bootstrapped-pytest/default.nix

```diff
@@ -0,0 +1,147 @@
+# A pytest(-xdist) package that does not depend on any other python packages
+# Can be used to test all other packages without causing infinite recursion
+
+{ bootstrapped-pip
+, fetchPypi
+, isPy27
+, lib
+, makeWrapper
+, python
+, stdenv
+, unzip
+
+# python packages to take sources from
+, apipkg
+, attrs
+, execnet
+, iniconfig
+, packaging
+, pluggy
+, py
+, pyparsing
+, pytest
+, pytest-forked
+, pytest-xdist
+, setuptools-scm
+, setuptools
+, toml
+, python2Packages
+, breakpointHook
+}:
+
+let
+
+  # the order is relevant for renaming unpacked archives
+  allDeps = [
+    apipkg
+    attrs
+    execnet
+    iniconfig
+    packaging
+    pluggy
+    py
+    pyparsing
+    pytest
+    pytest-forked
+    pytest-xdist
+    setuptools-scm
+    setuptools
+    toml
+  ] ++ (lib.optionals isPy27 ( with python2Packages; [
```

Comment:
Probably cleaner to make these function parameters (`pkg ? null`). Depending on how one overrides the package set this may fail now I think. E.g., doing some overriding of the `python2Packages` set will not affect the above packages, so they may still use an old version of the source.